lib/src/game/base_game.dart

```diff
@@ -125,6 +129,15 @@ class BaseGame extends Game with FPSCounter {
       c.onRemove();
       components.remove(c);
     });
+
+    if (this is HasCollidables) {
+      final collidables = (this as HasCollidables).collidables;
+      _removeLater.whereType<Collidable>().forEach((c) {
+        collidables.remove(c);
+      });
+      collidables.addAll(_addLater.whereType<Collidable>());
```

Comment:
Just wondering if this `collidables` property should be public, can the user break the collision detection system if they mess with this property?

Reply:
@spydon I think we should make `_collidables` private to the mixin and move this code inside.
so this becomes something like `if (this is HasCollidable) (this as HasCollidable).handleCollisions()` or something

Reply:
Missed this comment before, fixed now!